src/middleware/kv-static.ts

```diff
@@ -1,17 +1,45 @@
 import { Middleware } from '../Router'
 import { KVNamespace } from '@cloudflare/workers-types'
+const cache = { put: (path: string, num: number) => {} }
 
-interface KVStaticOptions {
-  kv: KVNamespace
-  maxAge?: number
+type CacheOptions = {
+  browserTTL: number
+  edgeTTL: number
+  bypassCache: boolean
+}
+type KVSiteOptions = {
+  /* handle requests that don't exist on the bucket */
+  notFoundHandler: (req: Request) => Response | Promise<Response>
+  /* configure how the incoming request's path is found in the bucket */
+  keyModifier: (path: string) => string
+  /* control the cache for all the site's content or on per request bias */
+  cacheControl: Partial<CacheOptions> | ((req: Request) => Partial<CacheOptions>)
+}
+type KVInit = {
+  /* Global KV namespace that is bound to the Worker script */
+  kv: Pick<KVNamespace, 'get'>
+  options: Partial<KVSiteOptions>
+}
```

Comment:
Let's keep all of these options in a single interface. Also, let's map `maxAge?: number` to the appropriate new option. That'll make all of these changes backwards compatible 👍